### ISSUE TITLE
[Bug] backword NaN when using flash_attn_varlen_qkvpacked_func

### ISSUE THREAD

**Issue by wangjunchao1118** [Dao-AILab/flash-attention#752]
Hi, I'm using Flashv2 with 2.4.2, i found a strange bug,  there is a NaN bug occur when i use the  flash_attn_varlen_qkvpacked_func.  

i have save the input "qkv" matrix to disk named "qkv.pt" 
[qkv.pt.zip](https://github.com/Dao-AILab/flash-attention/files/13828947/qkv.pt.zip)

thanks for your helps.

Here is a simple repro code:
`
# Tested with
# flash-attn-2.4.2
# torch==2.1.2
# NVIDIA A100-SXM4-80GB
# cuda 12.1
import torch
import flash_attn
import pdb
from flash_attn import flash_attn_varlen_qkvpacked_func

torch.manual_seed(1234)
qkv = torch.load('qkv.pt')
cu_seqlens = torch.IntTensor([0, 137, 486]).to(qkv.device)

if qkv.isnan().any():
    pdb.set_trace()

qkv.requires_grad_(True)

for i in range(5):
    grad = torch.ones([698, 16, 64], dtype=torch.bfloat16, device="cuda")
    y = flash_attn_varlen_qkvpacked_func(
        qkv,
        cu_seqlens,
        2048,   
        0.1,
        softmax_scale=None,
        causal=True,
        alibi_slopes=None,
    )
    print(i)
    # if i == 4:
    #     pdb.set_trace()
    if y.isnan().any():
        pdb.set_trace()
    y.backward(grad)
    print("QKV gradient:", "NaNs!" if qkv.grad.isnan().any().item() else "OK")
`

**Comment by Infi-zc** [comment]
@wangjunchao1118 hi, junchao, has this issue been resolved?
